### ISSUE TITLE
Does Min have a way to manage bookmarks ?   

### ISSUE THREAD

**Issue by id042013** [minbrowser/min#464]
I dont know how to access all bookmarks or if there is any way to manage all of them in Min.
It would be nice to have a feature to add tags. Its the only useful feature I seek from a browser

_The only reason I stopped using firefox is because it seems it cant handle well over 100.000 bookmarks. I looked into the firefox database and it seems that it uses the same table (moz_places) for bookmarks and also for tags. It would be cool to be able to import those into Min._

Help wanted, Feature request, Enhancements

**Comment by code-hunger** [comment]
`!bookmarks` in the searhbar, hit space, viola!
Unfortunately bookmarks can't be really grouped or sorted at the moment (AFAIK), but you can still search through them.

**Comment by PalmerAL** [comment]
Yeah, there isn't much in terms of bookmarking features right now. In terms of the specific features you mentioned, tagging definitely sounds doable, importing bookmarks from other browsers is fairly difficult but would be possible (I think there's an issue for this somewhere). My guess is that Min wouldn't work very well with 100k bookmarks either, although I haven't actually tested it (and if that turns out to be an issue, we could look into optimizing it).

**Comment by PalmerAL** [comment]
Also, we should probably add a menu item that maps to the !bookmarks shortcut, since it's not very obvious that you can do that.

**Comment by id042013** [comment]
Not necesarily need to develop an import bookmarks feature or to group or to sort bookmarks, advanced users could import it into SQL table if possible
But to become a permanent browser for me I'd need tags, (one sql table for bookmarks, and one for tags). Or I'll make an app when I have time only for bookmarks-with-tags that does not rely on any browser.  I dint find any good bookmarking app,( except defunct google toolbar). 

About those tags, 50.000 are gmail and google, I'll have to delete half from the total but its still big.  I just need to know what kind of database Min uses, in case it does, advanced users could import bookmarks and tags.

**Comment by PalmerAL** [comment]
Min uses IndexedDB (using <a href="https://github.com/dfahlander/Dexie.js">Dexie.js</a>) to store bookmarks. Here's the database schema: https://github.com/minbrowser/min/blob/a05c6b21049631a7fd6ed4b4b7266e71e2db915b/js/util/database.js#L21-L34

**Comment by id042013** [comment]
So this is a NoSQL database ?  Imports would be even easier, I guess. And each bookmark would have it's own tags ?

**Comment by PalmerAL** [comment]
@id042013 Kind of. There's a predefined schema like an SQL database, but it really behaves more like a NoSQL one (Internally, Chromium is using LevelDB (http://leveldb.org/), which is a NoSQL database). You can read more about IndexedDB here: https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API .

There isn't a field for tags right now, so you would have to add a field to the schema for that, and then modify the UI to search through that field and display the results somewhere.

Honestly, based on what you're looking for, I kind of doubt that Min is going to be able to do what you want. It would require a bunch of work to modify the UI to support tags, and I doubt the performance with that many bookmarks is going to be very good.

**Comment by franco27** [comment]
Hello 
I agree with the other users. We should be able to access the bookmarks.
It is very unfortunate that this function is not proposed.
I just discovered your web browser.

**Comment by PalmerAL** [comment]
You can view your bookmarks by typing !bookmarks into the search bar. What other management features would you like to see?

**Comment by franco27** [comment]
a print preview and a module to fill in the forms 
as SimpleFill (by context menu allows you to select an entry)
https://chrome.google.com/webstore/detail/simplefill/pjdkmeabgpggjenaichdfjhdckcejjik
or inFormEnter (an icon in the input fields and select the text)
https://chrome.google.com/webstore/detail/informenter%20/becedogggoaicbphcoimgpmhodofcfck
 
would also be appreciated, thank you

**Comment by PalmerAL** [comment]
@franco27 I've created a new issue for that: #563.

**Comment by franco27** [comment]
thank you
an entry "See his bookmarks" in the Min menu would be nice

**Comment by PalmerAL** [comment]
@franco27 Good idea, I've added that in 69816cad576b01b17d2432f98ba7e90b109baa30.
